sql/src/main/java/io/crate/operation/collect/MapSideDataCollectOperation.java

```diff
@@ -136,18 +131,7 @@ public MapSideDataCollectOperation(ClusterService clusterService,
         } else {
             CollectSource service = collectSourceResolver.getService(collectPhase, localNodeId);
             collectPhase = normalize(collectPhase);
-
-            if (collectPhase.whereClause().noMatch()) {
-                downstream.registerUpstream(this).finish();
-                return ImmutableList.of();
-            }
-            Collection<CrateCollector> collectors = service.getCollectors(collectPhase, downstream, jobCollectContext);
-            try {
-                launchCollectors(collectPhase, collectors);
-            } catch (EsRejectedExecutionException | RejectedExecutionException e) {
-                downstream.registerUpstream(this).fail(e);
-            }
-            return collectors;
+            return service.getCollectors(collectPhase, downstream, jobCollectContext);
```

Comment:
can you add an assertion here that getCollectors isn't empty ? or maybe in the launchCollectors. Because in case some source returns an empty list for whatever reason the query would probably hang as nobody would call finish on its downstreams